Sources/SwiftKuery/QueryBuilder.swift

```diff
@@ -18,24 +18,57 @@ import Foundation
 
 // MARK: QueryBuilder
 
-/// Used in code dealing with variances between the various database engines. As
-/// different databases have different query syntax, sometimes changes need to be
-/// made when generating the actual SQL statement to run. Additional needed changes
-/// should be done by updating QueryBuilder substitutions array. Every query component
-/// then builds its string representation using that array.
+/**
+ Used in code dealing with variances between the various database engines. As
+ different databases have different query syntax, sometimes changes need to be
+ made when generating the actual SQL statement to run. Additional needed changes
+ should be done by updating QueryBuilder substitutions array. Every query component
+ then builds its string representation using that array.
+ ### Usage Example: ###
+ In this example, a `QueryBuilder` for PostgreSQL is initialized.
+ Parameters are set for postgreSQL and Strings keywords for SQL queries are substituted into the queryBuilder.
+ The queryBuilder is then used to create a String description of an instance of the `Table` class called "todotable".
+ ```swift
+ let queryBuilder = QueryBuilder(withDeleteRequiresUsing: true, withUpdateRequiresFrom: true, createAutoIncrement: createAutoIncrement)
+ queryBuilder.updateSubstitutions([QueryBuilder.QuerySubstitutionNames.ucase : "UPPER",
+    QueryBuilder.QuerySubstitutionNames.lcase : "LOWER",
+    QueryBuilder.QuerySubstitutionNames.len : "LENGTH",
+    QueryBuilder.QuerySubstitutionNames.numberedParameter : "$",
+    QueryBuilder.QuerySubstitutionNames.namedParameter : "",
+    QueryBuilder.QuerySubstitutionNames.double : "double precision"
+ ])
+ let description = try todotable.build(queryBuilder: queryBuilder)
+ ```
+ */
 public class QueryBuilder {
+    // MARK: Substitutions
     /// An array of substitutions to be made in query String representation.
     public var substitutions: [String]
     
-    /// The index for the substitutions array.
+    /**
+     Enum defining the cases and their index for the substitutions array used by `QueryBuilder` to account for variances between the various database engines.
+     ### Usage Example: ###
+     In this example, a `QueryBuilder` for PostgreSQL is initialized.
+     Parameters are set for postgreSQL and Strings keywords for SQL queries are substituted into the queryBuilder. The `QuerySubstitutionNames` enum is used to refer to substitutions by name instead of their position in the "substitutions" array.
```

Comment:
can we change - "Strings keywords for SQL queries" to "string keywords for SQL queries"

Reply:
can we change "The queryBuilder is then used to create a String description of an instance of the Table class called todotable." to
"The `queryBuilder` is then used to create a String description of the Table class instance called `todotable`."